### DISCUSSION TITLE
Controller cutom dir not working

### DISCUSSION THREAD

**Discussion by riccycastro** [adonisjs/core#3358] [Help] [answered]
I don't want to have my controllers inside app/Controllers/Http/ instead they should be in a subfolder like this: app/Auth/Presentation/Http/AuthController, but I'm not having success on that.

I've found this previous answer, [here](https://github.com/adonisjs/core/issues/149#issuecomment-214297003) @thetutlage suggest adding the complete namespace, which I've tried and didn't work.

Controller location in my current app
![Screenshot from 2021-11-23 16-49-54](https://user-images.githubusercontent.com/14272438/143068699-73523eb3-2b42-4a63-ba0d-3e37caa72079.png)

Route setting
![Screenshot from 2021-11-23 16-49-30](https://user-images.githubusercontent.com/14272438/143068781-40d9c0e5-3f04-4275-a3de-9e60d5e1b21c.png)

Displayed error
![Screenshot from 2021-11-23 16-55-52](https://user-images.githubusercontent.com/14272438/143069072-5c6d95a6-851d-4f7d-96c1-c8323d64ad71.png)

it seems like Adonisjs requires that controllers must be inside Http folder.

How can I change that?

Best regards,
RIcardo Castro

**Comment by thetutlage** [comment, marked as answer]
Check the docs for controller namespaces https://docs.adonisjs.com/guides/controllers#route-namespacing

**Reply by riccycastro** [reply]
I was searching on route documentation :(...
Thank you very much
